hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncNonMetaRegionLocator.java

```diff
@@ -34,6 +34,7 @@
 import java.util.concurrent.ExecutionException;
```

Comment:
So the fix is either on SplitTableRegionProcedure (your suggestion), or UnassignProcedure (my proposal). In either case, shouldn't the added test go to the related "fixed" class, for consistency?

Reply:
Since this is mainly a condition triggered by a split operation, decided to move the test to _TestSplitOrMergeStatus_. Also renamed original test to _testSplitRegionReplicaRitRecovery_.